controller/status.go

```diff
@@ -0,0 +1,90 @@
+package controller
+
+import (
+	"fmt"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	appsv1alpha1 "github.com/openshift-console/console-application-operator/api/v1alpha1"
+)
+
+// SetDegraded sets the Operator Degraded condition with the provided reason and message.
+func SetDegraded(operatorCR *appsv1alpha1.ConsoleApplication, reason, message string) {
+	meta.SetStatusCondition(&operatorCR.Status.Conditions, metav1.Condition{
+		Type:               appsv1alpha1.ConditionOperatorDegraded.String(),
+		Status:             metav1.ConditionTrue,
+		Reason:             reason,
+		LastTransitionTime: metav1.NewTime(time.Now()),
+		Message:            message,
+	})
+}
+
+// SetGitServiceCondition sets the GitService condition with the provided status and reason.
+func SetGitServiceCondition(operatorCR *appsv1alpha1.ConsoleApplication, status metav1.ConditionStatus, reason string) {
+	meta.SetStatusCondition(&operatorCR.Status.Conditions, metav1.Condition{
+		Type:               appsv1alpha1.ConditionGitRepoReachable.String(),
+		Status:             status,
+		Reason:             reason,
+		LastTransitionTime: metav1.NewTime(time.Now()),
+		Message:            fmt.Sprintf("Git Repository Reachable: %s", string(status)),
+	})
+}
+
+// SetStarted sets the Operator Progressing condition to True.
+func SetStarted(operatorCR *appsv1alpha1.ConsoleApplication) {
+	meta.SetStatusCondition(&operatorCR.Status.Conditions, metav1.Condition{
+		Type:               appsv1alpha1.ConditionProgressing.String(),
+		Status:             metav1.ConditionTrue,
+		Reason:             "Reconciling",
+		LastTransitionTime: metav1.NewTime(time.Now()),
+		Message:            "Requirements are being met",
+	})
+}
+
+// SetFailed sets the Operator Progressing and Application Ready conditions to False with the provided reason and message.
+func SetFailed(operatorCR *appsv1alpha1.ConsoleApplication, reason, message string) {
+	meta.SetStatusCondition(&operatorCR.Status.Conditions, metav1.Condition{
+		Type:               appsv1alpha1.ConditionReady.String(),
+		Status:             metav1.ConditionFalse,
+		Reason:             reason,
+		LastTransitionTime: metav1.NewTime(time.Now()),
+		Message:            message,
+	})
+	meta.SetStatusCondition(&operatorCR.Status.Conditions, metav1.Condition{
+		Type:               appsv1alpha1.ConditionProgressing.String(),
+		Status:             metav1.ConditionFalse,
+		Reason:             appsv1alpha1.ReasonReconcileFailed.String(),
+		LastTransitionTime: metav1.NewTime(time.Now()),
+		Message:            message,
+	})
```

Comment:
I'm wondering if we really need this condition ?

Reply:
The idea was to have a condition that would give the real-time status of the progress of the ConsoleApplication; i.e. whether is it still creating the dependant resources (Builds, Deployments, etc.)

Reply:
The alternative approach may be to use the "Ready" type instead. 
Currently, I am setting "Ready" to True if all succeeds finally. If anything fails in between, I am setting "Ready" to false.
Maybe instead of having the "Progressing" condition type, I can set the "Ready" status to "Unknown" at the beginning of the loop.
Then user can infer that, if the "Ready" condition is "Unknown", it means the reconciliation is still going on.

Wdyt?

Reply:
Moving forward with this approach, as having the "Progressing" status condition is causing infinite reconciliation due to the way we are setting the status at regular interval, rather than setting it at the end.

Reply:
TODO: Implement "Progressing" status condition in the future.